sympy/physics/continuum_mechanics/beam.py

```diff
@@ -1561,6 +1577,159 @@ def plot_loading_results(self, subs=None):
+        """
+
+        Helper function for solve_for_ild_reactions(). It takes the unsubstituted
+        copy of the load equation and uses it to calculate shear force and bending
+        moment equations.
+        """
+
+        x = self.variable
+        shear_force = -integrate(self._original_load, x)
+        bending_moment = integrate(shear_force, x)
+
+        return shear_force, bending_moment
+
+    def solve_for_ild_reactions(self, val, *reactions):
+        """
+
+        Determines the Influence Line Diagram equations for reaction
+        forces under the effect of a moving load and returns a dictionary.
+
+        Parameters
+        ==========
+        val : Integer
+            Magnitude of moving load
+        reactions :
+            The reaction forces applied on the beam.
+
+        Examples
+        ========
+
+        There is a beam of length 10 meters. There are two simple supports
+        below the beam, one at the starting point and another at the ending
+        point of the beam. Calculate the I.L.D. equations for reaction forces
+        under the effect of a moving load of magnitude 1kN.
+
+        .. image:: ildreaction.png
+
+        Using the sign convention of downwards forces being positive.
+
+        .. plot::
+            :context: close-figs
+            :format: doctest
+            :include-source: True
+
+            >>> from sympy import symbols
+            >>> from sympy.physics.continuum_mechanics.beam import Beam
+            >>> E, I = symbols('E, I')
+            >>> R_0, R_10 = symbols('R_0, R_10')
+            >>> b = Beam(10, E, I)
+            >>> b.apply_support(0, 'roller')
+            >>> b.apply_support(10, 'roller')
+            >>> b.solve_for_ild_reactions(1,R_0,R_10)
+            {R_0: x/10 - 1, R_10: -x/10}
+
+        """
+        shear_force, bending_moment = self._solve_for_ild_reactions()
+        x = self.variable
+        l = self.length
+        C3 = Symbol('C3')
+        C4 = Symbol('C4')
+
+        shear_curve = limit(shear_force, x, l) - val
+        moment_curve = limit(bending_moment, x, l) - val*(l-x)
+
+        slope_eqs = []
+        deflection_eqs = []
+
+        slope_curve = integrate(bending_moment, x) + C3
+        for position, value in self._boundary_conditions['slope']:
+            eqs = slope_curve.subs(x, position) - value
+            slope_eqs.append(eqs)
+
+        deflection_curve = integrate(slope_curve, x) + C4
+        for position, value in self._boundary_conditions['deflection']:
+            eqs = deflection_curve.subs(x, position) - value
+            deflection_eqs.append(eqs)
+
+        solution = list((linsolve([shear_curve, moment_curve] + slope_eqs
+                            + deflection_eqs, (C3, C4) + reactions).args)[0])
+        solution = solution[2:]
+
+        # Determining the equations and solving them.
+        self._ild_reactions = dict(zip(reactions, solution))
+        return self._ild_reactions
```

Comment:
`solve_for_ild_reactions` probably shouldn't return anything, it should just update the value of `self._ild_reactions`.
It goes with the behavior of the other solve functions in the module.